Code_Exercises/9_sycl_of_hell/3_nd_range.cpp

```diff
@@ -0,0 +1,57 @@
+#include "argparse.hpp"
+#include <sycl/sycl.hpp>
+
+int main(int argc, char **argv) {
+
+  //  _                ___
+  // |_) _. ._ _  _     |  ._  ._     _|_
+  // |  (_| | _> (/_   _|_ | | |_) |_| |_
+  //
+  argparse::ArgumentParser program("3_nd_range");
+
+  program.add_argument("-g", "--global")
+      .help("Global Range")
+      .default_value(1)
+      .action([](const std::string &value) { return std::stoi(value); });
+
+  program.add_argument("-l", "--local")
+      .help("Local Range")
+      .default_value(1)
+      .action([](const std::string &value) { return std::stoi(value); });
+
+  try {
+    program.parse_args(argc, argv);
+  } catch (const std::runtime_error &err) {
```

Comment:
Should this be a `std::exception` instead of `runtime_error`? `argparse` lib throws things like `std::invalid_argument` which inherits from `std::exception`, not `runtime_error`, to my knowledge.

Reply:
No clue to be honst  
I just know that if I pass a bad argument, the `catch` is triggered and that if I remove the `try / catch` the error is 
```
applenco@x1921c0s2b0n0:~/sycltrain/9_sycl_of_hell> ./2_parallel_for -asdasdasd
terminate called after throwing an instance of 'std::runtime_error'
  what():  Unknown argument: -asdasdasd
Aborted
```